packages/actor-context-preprocess-statistic-link-discovery/lib/StatisticLinkDiscovery.ts

```diff
@@ -0,0 +1,84 @@
+// eslint-disable-next-line import/no-nodejs-modules
+import { EventEmitter } from 'events';
```

Comment:
Since the name of this package clashes with a node built-in (it's missing from your package.json's), we'll always have linter issues with it.
Maybe it's not worth it to use it, and just implement a basic listener system ourselves.
We can probably implement a `StatisticBase` or something similar for this.

Reply:
Right, which could also have typed events, as I believe is done in CSS. Should I create a separate package for this or where do you prefer to put it?

Reply:
I would start by implementing it here, and if needed, we can move it to a separate dedicated package.
We don't need anything too complicated here I think.

Reply:
This actor only tracks discover statistics, so we would create a between-actor dependency for other statistics (like link dereferences). I can start with that for now, but it would need to be refactored in the future.

Reply:
I implemented the StatisticLinkDiscovery with the StatisticBase you proposed. Is this what you meant? If so I can place the StatisticBase in `bus-context-preprocess` as all statistic tracker actors will use this package.

Reply:
I have some suggestions on class/method naming, but the idea looks good to me!

> If so I can place the StatisticBase in bus-context-preprocess as all statistic tracker actors will use this package.

This bus might not be the best suited place for this. But I don't have a better suggestion atm, so let's start with that for now, and move later if good inspiration arrives.